### PR TITLE
[JUJU-4284] Add cloud credential domain service and migration

### PR DESCRIPTION
**Pull request by wallyworld**
This PR adds the `domain/credential` service and import/export support. The service makes use of the previously added state support. The next PR will wire up the service into the API facades.

## QA steps

Run unit tests - cannot yet test end to end.

### CODE REVIEW

File: domain/credential/service/service.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	) (watcher.NotifyWatcher, error)
+}
+
+// State describes retrieval and persistence methods for credentials.
+type State interface {
+	// UpsertCloudCredential adds or updates a cloud credential with the given name, cloud, owner.
+	UpsertCloudCredential(ctx context.Context, name, cloudName, owner string, credential cloud.Credential) error
+
+	// InvalidateCloudCredential marks a cloud credential with the given tag as invalid.
+	//InvalidateCloudCredential(ctx context.Context, name, cloudName, owner, reason string) error
```
Comment (hpidcock) [inline domain/credential/service/service.go:31]: Add a TODO comment?

**Review by hpidcock — APPROVED** [review]
LGTM but seems to be some test fallout O_O

File: domain/credential/service/service.go (hunk 1 above)
Comment (wallyworld) [reply domain/credential/service/service.go]: removed till next PR

**Pull request merged**